app/src/main/kotlin/de/markusfisch/android/binaryeye/actions/wifi/WifiNetworkInfoFactory.kt

```diff
@@ -0,0 +1,41 @@
+package de.markusfisch.android.binaryeye.actions.wifi
+
+object WifiNetworkFactory {
+    private val wifiRegex = Regex("^WIFI:(T:(WEP|WPA|nopass))?;S:(.*);(P:(.*))?;(H?);$")
+    private val hexRegex = Regex("^[0-9a-f]*$", RegexOption.IGNORE_CASE)
+
+    fun parse(data: String): WifiNetworkInfo? {
+        return wifiRegex.matchEntire(data)?.groupValues?.let {
+            val security = it[2].let {
+                if (it == "" || it == "nopass") return null
```

Comment:
Correct me if I'm wrong, but doesn't this if return `null` for the `parse` function and breaks the parsing by doing that?

Reply:
Nope, it returns `null` from the `let` block, which is why the no-password tests still work :)